File: lib/src/application/commands/delete_foods_by_ids.dart
import 'package:functionally/reader_task.dart';
import 'package:functionally/task.dart';

typedef DeleteFoodsByIdsReader<ENV> = ReaderTask<ENV, void> Function(
  Set<String> ids,
);

typedef DeleteFoodsByIds = Task<void> Function(
  Set<String> ids,
);

File: src/application/commands/delete-foods-by-ids.ts
import { ReaderTaskEither } from 'fp-ts/ReaderTaskEither'
import { TaskEither } from 'fp-ts/TaskEither'

export type DeleteFoodsByIds = (
	ids: ReadonlySet<string>
) => TaskEither<Error, void>

export type DeleteFoodsByIdsWithDeps<ENV> = (
	ids: ReadonlySet<string>
) => ReaderTaskEither<ENV, Error, void>
